src/Environment.cc

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <ignition/utils/Environment.hh>
+
+/////////////////////////////////////////////////
+bool ignition::utils::env(const std::string &_name,
+                          std::string &_value,
+                          bool _allowEmpty)
+{
+  std::string v;
+  bool valid = false;
+#ifdef _WIN32
+  const DWORD buffSize = 32767;
+  static char buffer[buffSize];
```

Comment:
Does the fact that we're using a static char buffer mean this is not thread safe?

Reply:
Removed in 2fb0b28